modules/infrastructure-vsphere-failure-domains-yaml.adoc

```diff
@@ -0,0 +1,99 @@
+        - datacenters: <2>
+            - region-a-dc
+            - region-b-dc
+          port: 443 <3>
+          server: your.vcenter.server <4>
+      failureDomains: <5>
+        - name: failure-domain-1 <6>
+          region: region-a <7>
+          zone: zone-a <8>
+          server: your.vcenter.server <9>
+          topology: <10>
+            datacenter: region-a-dc <11>
+            computeCluster: "/region-a-dc/host/zone-a-cluster" <12>  
+            resourcePool: "/region-a-dc/host/zone-a-cluster/Resources/resource-pool" <13> 
+            datastore: "/region-a-dc/datastore/datastore-a" <14>
+            networks: <15>
+            - port-group
+        - name: failure-domain-2
+          region: region-a
+          zone: zone-b
+          server: your.vcenter.server
+          topology:
+            computeCluster: /region-a-dc/host/zone-b-cluster
+            datacenter: region-a-dc
+            datastore: /region-a-dc/datastore/datastore-a            
+            networks:
+            - port-group            
+        - name: failure-domain-3
+          region: region-b
+          zone: zone-a
+          server: your.vcenter.server
+          topology:
+            computeCluster: /region-b-dc/host/zone-a-cluster
+            datacenter: region-b-dc
+            datastore: /region-b-dc/datastore/datastore-b
+            networks:
+            - port-group       
+      nodeNetworking:
+        external: {}
+        internal: {}
+----
+
+<1> The list of vCenter servers associated with the {product-title} cluster. Only one vCenter may be defined.
+
+<2> The list of vCenter datacenters where VMs associated with the {product-title} cluster will be created or presently exist.
+
+<3> The TCP port of the vCenter server.
+
+<4> The FQDN of the vCenter server.
+
+<5> The list of failure domains.
+
+<6> The name of the failure domain. 
+
+<7> The value of the `openshift-region` tag assigned to the topology for the failure failure domain.
+
+<8> The value of the `openshift-zone` tag assigned to the topology for the failure failure domain.
+
+<9> The name of the vCenter server as defined by <4>
+
+<10> The vCenter reources associated with the failure domain.
+
+<11> The datacenter associated with the failure domain. 
+
+<12> The full path of the compute cluster associated with the failure domain. 
+
+<13> Optional: The full path of the resource pool associated with the failure domain. 
+
+<14> The full path of the datastore associated with the failure domain. 
+
+<15> A list of port groups associated with the failure domain. Only one portgroup may be defined.
+
+
+You can edit the `Infrastructure` CRD instance containing the topology configuration by running the following command:
+[source,terminal]
+----
+$ oc edit infrastructures.config.openshift.io cluster
+----
+
+[IMPORTANT]
+====
+Once a failure domain has been created it must not be deleted or modified.  However, new failure domains can be appended to the list of failure domains.
```

Comment:
Do we plan to enumerate the things you must do before doing this? Would it help the end user to understand if we had a list of, here's what would break/here's what you'd need to do if you _really_ wanted to remove one?

Statements like this always seem weird to me when they don't have justifications for why

Reply:
Thanks Joel.  @gnufied would you mind providing a quick summary of the 'why' and I can incorporate it in to the PR?

Reply:
Sure, but some follow up questions. Is this storage only item? Say for example - if adding a failure-domain results in creation of VMs in that failure-domain , will removal of failure-domain result in deletion of VMs in that failure-domain? (i.e will get implemented?).